controllers/namespace_updater.go

```diff
@@ -12,7 +12,9 @@ import (
 func (r *OpenshiftServiceMeshReconciler) addGatewayAnnotations(ctx context.Context, namespace *v1.Namespace) error {
 	if namespace.ObjectMeta.Annotations[AnnotationPublicGatewayExternalHost] != "" &&
 		namespace.ObjectMeta.Annotations[AnnotationPublicGatewayInternalHost] != "" &&
-		namespace.ObjectMeta.Annotations[AnnotationPublicGatewayName] != "" {
+		namespace.ObjectMeta.Annotations[AnnotationPublicGatewayName] != "" &&
+		namespace.ObjectMeta.Annotations[AnnotationProjectModelGatewayHostPatternExternal] != "" &&
+		namespace.ObjectMeta.Annotations[AnnotationProjectModelGatewayHostPatternInternal] != "" {
```

Comment:
It's getting a bit hard to read now. Maybe we can have a func such as 

```
func isAnnotated(objectMeta, annotations... string) bool
```

which will improve readability?